File: androidClient/app/src/main/java/com/macbitsgoa/comrades/persistance/Course.java
package com.macbitsgoa.comrades.persistance;

import com.google.firebase.database.Exclude;

import javax.annotation.Nullable;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Ignore;

import static androidx.room.ForeignKey.RESTRICT;

/**
 * Definition of table course.
 *
 * @author Rushikesh Jogdand.
 */
@Entity(primaryKeys = {"id"},
        foreignKeys = @ForeignKey(entity = Person.class,
                parentColumns = "id",
                childColumns = "addedById",
                onDelete = RESTRICT,
        onUpdate = 1))
public class Course {
    @NonNull
    public String name;

    @NonNull
    public String code;

    @NonNull
    public String id;

    @NonNull
    public String addedById;

    // Handy attribute. You will have to populate this manually.
    @Ignore
    @Exclude
    public LiveData<Person> addedBy;

    @Exclude
    public boolean isFollowing;
}

File: androidClient/app/src/main/java/com/macbitsgoa/comrades/persistance/Course.kt
package com.macbitsgoa.comrades.persistance

import androidx.lifecycle.LiveData
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.RESTRICT
import androidx.room.Ignore
import com.google.firebase.database.Exclude

/**
 * Definition of table course.
 *
 * @author Rushikesh Jogdand.
 */
@Entity(primaryKeys = ["id"],
        foreignKeys = [ForeignKey(entity = Person::class,
                parentColumns = ["id"],
                childColumns = ["addedById"],
                onDelete = RESTRICT,
                onUpdate = 1)])
class Course {
    var name: String = ""

    var code: String = ""

    var id: String = ""

    var addedById: String = ""

    // Handy attribute. You will have to populate this manually.
    @Ignore
    @Exclude
    var addedBy: LiveData<Person>? = null

    @Exclude
    var isFollowing: Boolean = false
}
